src/runtime/image.ts

```diff
@@ -137,36 +137,61 @@ export function createImage (context, { providers, defaultProvider, presets, int
   return image
 }
 
-function createObserver (options: object) {
-  const intersectOptions = {
+function printObserver (onMatch) {
+  if (typeof window === 'undefined' || typeof window.matchMedia === 'undefined') {
+    return
+  }
+
+  const mediaQueryList = window.matchMedia('print')
+  mediaQueryList.addListener((query) => {
+    if (query.matches) {
+      onMatch()
+    }
+  })
+}
+
+function intersectionObserver (onMatch, options) {
+  const observer = (typeof IntersectionObserver !== 'undefined' ? new IntersectionObserver(onMatch, {
     rootMargin: '50px',
     ...options
-  }
-  const observer = (typeof IntersectionObserver !== 'undefined' ? new IntersectionObserver(callback, intersectOptions) : {}) as IntersectionObserver
+  }) : {}) as IntersectionObserver
+
+  return observer
+}
+
+function createObserver (intersectionOptions: object) {
```

Comment:
I think for longer term it would be better to create observer directly in component instead of share runtime logic (`createObserver(image)`)

Reply:
If we move observer into component we are creating multiple query match and IntersectionOsberver. I'm not sure but it may reduce the performance